### PR TITLE
WIP fixes

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

**Comment by Lemiru** [comment]
@Nekotekina With this PR i DMC HD Collection trying to launch games from launcher (with "Load liblv2.sprx only") results in
`F {PPU[0x1000000] Thread (main_thread) [0x00551dd4]} class std::runtime_error thrown: _sys_process_exit2`
On master (also with "Load liblv2.sprx only") It would just freeze on
`U {PPU[0x1000000] Thread (main_thread) [0x00551dd4]} HLE TODO: Unimplemented syscall _sys_process_exit -> CELL_OK`
I'm attaching logs for both cases
[Master.log.gz](https://github.com/RPCS3/rpcs3/files/1308631/Master.log.gz)
[WIP.log.gz](https://github.com/RPCS3/rpcs3/files/1308635/WIP.log.gz)

**Comment by SakataGintokiYT** [comment]
Some games shows this:
bez tytulu
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1309017/RPCS3.log.gz)

**Comment by hcorion** [comment]
Wait, is it my birthday? `_sys_look_ctype_table`, `_sys_process_exit` and general `_sys` improvements!?
Hooray!
All we need is `_sys_qsort` implementation and we'll be set!

File: rpcs3/Emu/Cell/Modules/sys_libc_.cpp [hunk 1]
```diff
@@ -277,13 +277,17 @@ vm::ptr<char> _sys_strncat(vm::ptr<char> dest, vm::cptr<char> source, u32 len)
...
 {
-	sysPrxForUser.trace("_sys_strcpy(dest=*0x%x, source=%s)", dest, source);
+	sysPrxForUser.trace("_sys_strcpy(dst=*0x%x, src=%s)", dst, src);
 
-	verify(HERE), std::strcpy(dest.get_ptr(), source.get_ptr()) == dest.get_ptr();
-
-	return dest;
+	for (u32 i = 0;; i++)
+	{
+		if (!(dst[i] = src[i]))
```
Comment (oobol2000oo) [inline rpcs3/Emu/Cell/Modules/sys_libc_.cpp]: It should be **if (!(dst[i] == src[i]))** right?

**Comment by Zangetsu38** [comment]
in last commit on MGS 4
>E {PPU[0x1000000] Thread (main_thread) [0x000474b0]} 'sceNpDrmProcessExitSpawn2' failed with 0x80010005 : CELL_ESRCH [1]
F {PPU[0x1000000] Thread (main_thread) [0x000474b0]} MEM: Access violation reading location 0x0

**Comment by Zangetsu38** [comment]
Good now fixed Thx

**Comment by Lemiru** [comment]
Now not only DMC HD Collection launcher is fixed but also we can finally play as Lucia. This PR is just beautiful.
screen shot 09-18-17 at 08 04 pm

**Comment by Zangetsu38** [comment]
Zone of the ender error same MGS 4 before
[last build appveyor RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1312260/RPCS3.log.gz)
>E {PPU[0x1000000] Thread (main_thread) [0x0026e310]} 'sceNpDrmProcessExitSpawn2' failed with 0x80010005 : CELL_ESRCH [1]
F {PPU[0x1000000] Thread (main_thread) [0x0026e310]} MEM: Access violation reading location 0x0

**Comment by kamer1337** [comment]
missing CELL_EINVAL in sys_fs_open 
switch case

**Comment by Zangetsu38** [comment]
ZOE and MGS 4 fixed again, ZOE LAUNCHER works, but not want boot second game, show windows and close directly with this error
[Last Appveyor RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1312577/RPCS3.log.gz)
>U {PPU[0x1000000] Thread (main_thread) [0x0088d754]} HLE TODO: Unimplemented syscall syscall_988 -> CELL_OK
U {PPU[0x1000000] Thread (main_thread) [0x02b9dc88]} sys_prx TODO: _sys_prx_get_module_list(flags=2, pInfo=*0xd000f0a8)

ZOE1 HD show only blank window gsrender
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1312580/RPCS3.log.gz)

I have found problem, need copied param.sfo in root folder USRDIR and both game start 
Collection hd have this problem, not first time i see that, need copied param.sfo for game works

rpcs3 launch self in "dev_hdd0\game\NPEB01186\USRDIR\ZoE1" and try open sfo but sfo in "dev_hdd0\game\NPEB01186\USRDIR\" for that, after sfo copied game boot and show one think, need improve sys fs for this

**Comment by Zangetsu38** [comment]
Improved also for launcher Ridge Racer 7, (Update) now game works in lv2 mode, but intro video causing problem, same few other game show same think in lv2 mode, and for this game lv2 mode is obligatory for launcher works 
[Last appveyor RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1312890/RPCS3.log.gz)
>F {PPU[0x1000013] Thread (HLE Video Decoder) [0x00c6afb0]} MEM: Access violation reading location 0xffff8fd4

File: rpcs3/Emu/Cell/Modules/sys_libc_.cpp [hunk 2]
```diff
@@ -129,82 +189,160 @@ vm::ptr<void> _sys_memmove(vm::ptr<void> dst, vm::cptr<void> src, u32 size)
...
+		}
+	}
 }
 
 s32 _sys_strcmp(vm::cptr<char> str1, vm::cptr<char> str2)
 {
 	sysPrxForUser.trace("_sys_strcmp(str1=%s, str2=%s)", str1, str2);
 
-	return std::strcmp(str1.get_ptr(), str2.get_ptr());
+	for (u32 i = 0;; i++)
```
Comment (henriquenj) [inline rpcs3/Emu/Cell/Modules/sys_libc_.cpp]: I'm trying to understand a bit of RPCS3 code. This change seems odd. What's the actual difference between using the ones from the standard system libraries and reimplementing? Are the system libraries inefficient?

Comment (hcorion) [reply rpcs3/Emu/Cell/Modules/sys_libc_.cpp]: I don't understand why, but it's a bad idea to pass virtual PS3 memory to system calls.

Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/sys_libc_.cpp]: Standard doesn't specify exact return values (-1 or 1 for example) and defines many undefined behavior conditions.

Comment (henriquenj) [reply rpcs3/Emu/Cell/Modules/sys_libc_.cpp]: I see, now it makes more sense. Thanks.

**Pull request merged**